filters/scheduler/lifo.go

```diff
@@ -286,24 +287,25 @@ func (l *lifoGroupFilter) Response(ctx filters.FilterContext) {
 	response(scheduler.LIFOKey, ctx)
 }
 
-func request(q *scheduler.Queue, key string, ctx filters.FilterContext) {
+func request(q *scheduler.Queue, key string, ctx filters.FilterContext, metricKey string) {
 	if q == nil {
 		log.Warningf("Unexpected scheduler.Queue is nil for key %s", key)
 		return
 	}
 
 	done, err := q.Wait()
 	if err != nil {
-		// TODO: replace the log with metrics
 		switch err {
 		case jobqueue.ErrStackFull:
 			log.Debugf("Failed to get an entry on to the queue to process QueueFull: %v for host %s", err, ctx.Request().Host)
+			ctx.Metrics().IncCounter(fmt.Sprintf("lifo.%s.rejected", metricKey))
```

Comment:
This implementation is basic and does not generate a huge diff. It is also possible to record these metrics in the `scheduler` package.
Benefits of doing that include the ability to create the metric keys once and binding the recording of the metrics to the option `EnableRouteLIFOMetrics`. The change would be large though.
Let me know if that is preferable.

Reply:
Hello, thank you for the contribution.
Indeed it looks like lifo queue already contains most of the infrastructure to collect metrics
https://github.com/zalando/skipper/blob/2059778e1924aaafb86accfc0b27e10245f13117/scheduler/scheduler.go#L190-L198
so it should be possible to collect these inside the `Wait`
https://github.com/zalando/skipper/blob/2059778e1924aaafb86accfc0b27e10245f13117/scheduler/scheduler.go#L126-L132
IMO that would be preferable because it would reside in one place, configurable by the option and metric key (queue name that is route id) would be uniform across metrics

Reply:
Thanks for the feedback @AlexanderYastrebov !
I've pushed the changes.